### DISCUSSION TITLE
How can I retrieve the wall clock time (mtime) from HDL gateware?

### DISCUSSION THREAD

**Discussion by mcoroyer** [stnolting/neorv32#761] [Q&A] [answered]
Hello,

I'm taking over a colleague's project based on neorv32, and I have just migrated it from v1.6.7 to v1.9.2.

However, I notice that "mtime_o" has been removed from the top entity.
We used this output to schedule and synchronize actions between gateware (HDL) and firmware (C).

Was this bad practice? How can I retrieve the wall clock time (mtime) from HDL gateware?

Thanks in advance!

**Comment by stnolting** [comment]
Hey there!

> However, I notice that "mtime_o" has been removed from the top entity.

Yeah, I remember that... It has been removed - but I'm not sure why this is the case 🙈

> Was this bad practice? How can I retrieve the wall clock time (mtime) from HDL gateware?

It's just fine. Actually, I still think this is a nice feature so it should be re-added. Would you like to make a PR for that?

**Comment by mcoroyer** [comment, marked as answer]
Great if this feature can be retained!
Thanks a lot! :)

#762
